File: NET/JNetCall.Sharp.Tests/IO/UdpTransportTest.cs
using System.Threading;
using JNetCall.Sharp.API.IO;
using JNetCall.Sharp.Impl.IO.Net;

namespace JNetCall.Sharp.Tests.IO
{
    public sealed class UdpTransportTest : TransportTest
    {
        private static int _offset;
        private static int NextOffset => Interlocked.Increment(ref _offset);

        protected override (ISendTransport, ISendTransport) GetBoth()
        {
            var offset = NextOffset;
            var portOne = 11001 + offset;
            var portTwo = 11051 + offset;
            var left = new UdpTransport(
                Encoding,
                "localhost", portOne,
                "localhost", portTwo
            );
            var right = new UdpTransport(
                Encoding,
                "localhost", portTwo,
                "localhost", portOne
            );
            return (left, right);
        }
    }
}

File: JVM/jnetcall-java-tests/src/test/java/jnetcall/java/tests/io/UdpTransportTest.java
package jnetcall.java.tests.io;

import java.util.concurrent.atomic.AtomicInteger;

import org.javatuples.Pair;

import jnetcall.java.api.io.ISendTransport;
import jnetcall.java.impl.io.net.UdpTransport;

public final class UdpTransportTest extends TransportTest {

    private static AtomicInteger _offset = new AtomicInteger();

    private static int getNextOffset() {
        return _offset.incrementAndGet();
    }

    @Override
    protected Pair<ISendTransport, ISendTransport> getBoth() {
        int offset = getNextOffset();
        int portOne = 11001 + offset;
        int portTwo = 11051 + offset;
        UdpTransport left = new UdpTransport(
                Encoding,
                "localhost", portOne,
                "localhost", portTwo
        );
        UdpTransport right = new UdpTransport(
                Encoding,
                "localhost", portTwo,
                "localhost", portOne
        );
        return Pair.with(left, right);
    }
}
